fc_util/src/lib.rs

```diff
@@ -3,57 +3,180 @@
 
 extern crate libc;
 
+use std::fmt;
+
 pub mod validators;
 
-pub fn timestamp_cycles() -> u64 {
-    #[cfg(target_arch = "x86_64")]
-    // Safe because there's nothing that can go wrong with this call.
-    unsafe {
-        std::arch::x86_64::_rdtsc() as u64
-    }
-    #[cfg(not(target_arch = "x86_64"))]
-    {
-        let mut ts = libc::timespec {
+const SECONDS_TO_NANOS: u64 = 1_000_000_000;
```

Comment:
Ah here's the `const`. Please replace other uses of the magic number `1_000_000_000` with `SECONDS_TO_NANOS` (might need to make it `pub` for that - in which case don't forget to properly document it)